### DISCUSSION TITLE
Completion shows me duplicate rows of file options

### DISCUSSION THREAD

**Discussion by grimm26** [zimfw/zimfw#491] [Q&A] [answered]
When I do tab completion, I get duplicate rows showing file options:
![image](https://user-images.githubusercontent.com/127401/203806768-51661fe7-e30c-4f9f-be9e-cec0bf87169d.png)

I'm using the zim completions modules, here is my whole [zimrc](https://github.com/grimm26/dotfiles/blob/main/.zimrc%23%23template). It's templated in yadm, so ignore the templating stuff :).

Is this happening to anyone else? Any ideas? I'm going to tinker with the completions module to see if I can figure it out.

**Comment by ericbn** [comment]
Does setting the completion case-sensitivity to sensitive makes the issue go away?

**Reply by grimm26** [reply]
![image](https://user-images.githubusercontent.com/127401/205444103-686c6054-ccbd-4557-97e5-26d305a9383c.png)
 

No, just stops including `Templates` in suggestions.

**Comment by grimm26** [comment]
ok, found some weird behavior depending on how I tab complete:
![image](https://user-images.githubusercontent.com/127401/205466391-4b3cb44b-1b11-48a8-bf89-e536d19978b1.png)
![image](https://user-images.githubusercontent.com/127401/205466403-fe5590a8-c68b-4b2d-a541-f6187bc3622a.png)

So when using `file` or `ls` I only get one copy of the completion descriptions, but with `ls` the FILE heading is in caps.

**Reply by grimm26** [reply]
even when I let zoxide be its default function of `z` instead of overrding `cd`, I still get the double completion results with `z`. If I remove zimfw from the equation, completion results are normal.

**Comment by grimm26** [comment]
ok, those differences gave me the hint. I use [zoxide](https://github.com/ajeetdsouza/zoxide) to replace the cd command and that is what is causing the weird completion behavior with `cd`.
Which brings me to a question of how to handle times when I have something like this that runs comdef commands. I have to run it after zim completions module is run in order to have the compdef command available. But if I then run `zimfw check-dumpfile` it will detect that it needs to rebuild the dumpfile and tells me to start a new shell. This can be done ad infinitum. I have the same issue within `~/.zimrc` with 
```
zmodule ohmyzsh/ohmyzsh --root plugins/aws                                
zmodule ohmyzsh/ohmyzsh --root plugins/git                                
zmodule ohmyzsh/ohmyzsh --root plugins/terraform                          
zmodule ohmyzsh/ohmyzsh --root plugins/ubuntu  
```
they need compinit already to have been run cuz the use comp* commands.

**Reply by ericbn** [reply]
`zimfw check-dumpfile` does not behave well with completions defined via comp* commands. This is a known issue.

**Reply by ericbn** [reply]
Improved `zimfw check-dumpfile` in version [1.11.1](https://github.com/zimfw/zimfw/releases/tag/v1.11.1), just released. It should consistently behave as expected now.

**Comment by grimm26** [comment, marked as answer]
I commented out all of the zstyle lines in the completion module and then started putting them back one by one. After this line is put back in, the duplicate results happen:
`zstyle ':completion:*' group-name ''`

I opened an issue on the zoxide project, https://github.com/ajeetdsouza/zoxide/issues/491
